phoenicis-library/src/main/java/org/phoenicis/library/ShortcutManager.java

```diff
@@ -39,42 +42,45 @@
     private final String shortcutDirectory;
     private final LibraryManager libraryManager;
     private final ScriptInterpreter scriptInterpreter;
+    private ObjectMapper objectMapper;
     private final String desktopShortcutDirectory;
 
-    ShortcutManager(String shortcutDirectory, String desktopShortcutDirectory, LibraryManager libraryManager,
-            ScriptInterpreter scriptInterpreter) {
+    ShortcutManager(String shortcutDirectory,
+            String desktopShortcutDirectory,
+            LibraryManager libraryManager,
+            ScriptInterpreter scriptInterpreter,
+            ObjectMapper objectMapper) {
         this.shortcutDirectory = shortcutDirectory;
         this.desktopShortcutDirectory = desktopShortcutDirectory;
         this.libraryManager = libraryManager;
         this.scriptInterpreter = scriptInterpreter;
+        this.objectMapper = objectMapper;
     }
 
     public void createShortcut(ShortcutDTO shortcutDTO) {
         final String baseName = shortcutDTO.getId();
         final File shortcutDirectoryFile = new File(this.shortcutDirectory);
 
-        final File nameFile = new File(shortcutDirectoryFile, baseName + ".name");
-        final File categoryFile = new File(shortcutDirectoryFile, baseName + ".category");
+        final File infoFile = new File(shortcutDirectoryFile, baseName + ".info");
         final File scriptFile = new File(shortcutDirectoryFile, baseName + ".shortcut");
         final File iconFile = new File(shortcutDirectoryFile, baseName + ".icon");
         final File miniatureFile = new File(shortcutDirectoryFile, baseName + ".miniature");
-        final File descriptionFile = new File(shortcutDirectoryFile, baseName + ".description");
 
         if (!shortcutDirectoryFile.exists()) {
             shortcutDirectoryFile.mkdirs();
         }
 
+        final Map<String, String> shortcutInfo = new HashMap<>();
+        shortcutInfo.put("name", shortcutDTO.getName());
+        shortcutInfo.put("category", shortcutDTO.getCategory());
+        shortcutInfo.put("description", shortcutDTO.getDescription());
+
         try {
+            String infoJson = this.objectMapper.writeValueAsString(shortcutInfo);
```

Comment:
I would like it better, if we had a wrapper object with some annotations instead of a `HashMap`. This would make it easier to track what attributes are contained inside an `info` file.
An example for what I mean are the `RepositoryLocation` classes

Reply:
Wouldn't that be almost duplicate of `ShortcutDTO`?

Reply:
Then I would suggest tomove the stuff from inside `ShortcutDTO` to the new class with annotations. 
Afterwards we can add an instance of the new class `ShortcutInfo` to `ShortcutDTO`.